torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1926,18 +1954,32 @@ def hint():
                 if r.upper != sympy.oo and r.upper < sys.maxsize - 1:
                     bounds.append(str(r.upper))
                 if len(bounds) > 1:
-                    exprs.append(" <= ".join(bounds))
+                    exprs.append(ShapeGuardExprSources(" <= ".join(bounds), guard_sources))
 
         if constraint_violations:
-            msgs = [f"  {i + 1}. {msg()}" for i, msg in enumerate(constraint_violations)]
-            msgs = "\n".join(msgs)
-            raise ConstraintViolationError(f"Constraints violated!\n{msgs}")
+            raise_msgs = []
+            warn_msgs = []
+            for msg_fn in constraint_violations:
+                constraint, msg = msg_fn()
+                if constraint.warn_only:
+                    msg = f"  {len(warn_msgs) + 1}. {msg}"
+                    warn_msgs.append(msg)
+                else:
+                    msg = f"  {len(raise_msgs) + 1}. {msg}"
+                    raise_msgs.append(msg)
+
+            if len(raise_msgs) > 0:
+                raise ConstraintViolationError(f"Constraints violated!\n Erroring constraints: \n {raise_msgs} \n Warning only constraints: \n {warn_msgs}")
+            else:
+                log.debug(f"Warning only constraints violated \n {warn_msgs}")
```

Comment:
Why is this a debug log

Reply:
A more logical structure for the code here would be to unconditionally `log.warning` the warning constraints, and then raise if there are any erroring constraints. This allows the warnings to be handled uniformly.

Reply:
sg